gateway-ha/src/main/java/io/trino/gateway/ha/module/HaGatewayProviderModule.java

```diff
@@ -101,6 +115,12 @@ public HaGatewayProviderModule(HaGatewayConfiguration configuration)
 
         OAuth2GatewayCookieConfigurationPropertiesProvider oAuth2GatewayCookieConfigurationPropertiesProvider = OAuth2GatewayCookieConfigurationPropertiesProvider.getInstance();
         oAuth2GatewayCookieConfigurationPropertiesProvider.initialize(configuration.getOauth2GatewayCookieConfiguration());
+
+        Jdbi jdbi = Jdbi.create(configuration.getDataStore().getJdbcUrl(), configuration.getDataStore().getUser(), configuration.getDataStore().getPassword());
+        JdbcConnectionManager connectionManager = new JdbcConnectionManager(jdbi, configuration.getDataStore());
+        resourceGroupsManager = new HaResourceGroupsManager(connectionManager);
+        gatewayBackendManager = new HaGatewayManager(jdbi);
+        queryHistoryManager = new HaQueryHistoryManager(jdbi, configuration.getDataStore().getJdbcUrl().startsWith("jdbc:oracle"));
```

Comment:
I think we should manage these classes in guice instead of initializing here as follow-up.